src/sonic-config-engine/portconfig.py

```diff
@@ -51,4 +111,90 @@ def parse_port_config_file(port_config_file):
             port_alias_map[data['alias']] = name
     return (ports, port_alias_map)
 
+# Generate configs (i.e. alias, lanes, speed, index) for port
+def gen_port_config(ports, intf, match_list, index, alias_at_lanes, lanes):
+    parent_interface_index = int(re.search("Ethernet(\d+)", intf).group(1))
+    alias_start = 0
+    lanes_start = 0
+    """
+    Example of match_list for some breakout_mode using regex
+        Breakout Mode -------> Match_list
+        -----------------------------
+        2x25G(2)+1x50G(2) ---> [('2', '25G', None, '(2)', '2'), ('1', '50G', None, '(2)', '2')]
+        1x50G(2)+2x25G(2) ---> [('1', '50G', None, '(2)', '2'), ('2', '25G', None, '(2)', '2')]
+        1x100G[40G] ---------> [('1', '100G', '[40G]', None, None)]
+        2x50G ---------------> [('2', '50G', None, None, None)]
+    """
+    for k in match_list:
+
+        # Logic for ASYMMETRIC breakout mode
+        if k[4] is not None:
+            step = int(k[4])/int(k[0])
+            for i in range(0,int(k[4]), step):
+                intf_name = PORT_STR + str(parent_interface_index)
+
+                ports[intf_name] = {}
+                ports[intf_name]['alias'] = alias_at_lanes.split(",")[alias_start]
+                ports[intf_name]['lanes'] = ','.join(lanes.split(",")[alias_start:alias_start+step])
+                ports[intf_name]["speed"] = k[1]
+                ports[intf_name]['index'] = index.split(",")[alias_start]
+
+                parent_interface_index += int(k[4])/int(k[0])
+                alias_start += step
 
+        # Logic for SYMMETRIC breakout mode
+        else:
+            lane_len = len(lanes.split(","))
+            step = lane_len/int(k[0])
+            lane_end = step
+            for i in range(0,lane_len, step):
+                intf_name = PORT_STR + str(parent_interface_index)
+
+                ports[intf_name] = {}
```

Comment:
Looks like we should be able to merge symmetric and asymmetric logic as most of the code looks similar.

Reply:
modified to one logic.